docs/integration-testing.md

```diff
@@ -0,0 +1,128 @@
+# Writing Integration Tests
+
+## Definition and distinction
```

Comment:
note that this does not match exactly other definitions, like then ones proposed in https://martinfowler.com/articles/microservice-testing/#testing-unit-introduction
Does not mean that we have to change it, just wanted to make sure that we're aware of that.